fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/RecordMetaDataBuilder.java

```diff
@@ -435,6 +444,53 @@ public RecordMetaDataBuilder setRecords(@Nonnull Descriptors.FileDescriptor file
         return this;
     }
 
+    /**
+     * Update the records descriptor of the record meta-data.
+     *
+     * <p>
+     * This involves adding new record types and updating descriptors for the existing record types and union fields.
+     * By contract, the extension options will be processed for all of the new record types and will not be processed
+     * for any of the old record types. Also, it is not allowed to call this method when the local file descriptor is set.
```

Comment:
Given that the other overload of this function now has more information in its Javadoc description, I feel like this should link to the other one.